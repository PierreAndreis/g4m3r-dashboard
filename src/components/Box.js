import React, { Component } from "react";
import { css } from "emotion";
import classNames from "classnames";

const BoxBase = css`
  background: white;
  ${"" /* border-radius: 15px; */};
  width: 350px;
  ${"" /* box-shadow: rgba(0, 0, 0, 0.05) 0px 6px 30px; */};
  ${"" /* box-shadow: 0 4px 9px -3px rgba(102, 136, 153, 0.15); */};
  box-shadow: 0 10px 40px 0 rgba(0, 0, 0, 0.1);
  border-radius: 3px;
  box-sizing: border-box;
  margin: 5px;
`;

const BoxCenter = css`
  margin: 0 auto;
  align-self: center;
  justify-self: center;
`;

const BoxPadding = css`
  padding: 15px 0;
`;

const BoxTitle = css`
  display: flex;
  font-size: 18px;
  color: #4a4a4a;
  margin: 0 10px 10px;
`;

const BoxBody = css`
  width: 100%;
  box-sizing: border-box;

  .${BoxTitle} {
    margin: 0;
  }

  .${BoxPadding} & {
    & > div {
      padding: 5px 15px;
    }
  }
`;

const BoxOption = css`
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-sizing: border-box;

  &:nth-child(odd) {
    background: rgba(0, 0, 0, 0.03);
  }
`;

export default class Box extends Component {
  static Title = ({ children, className, ...otherProps }) => (
    <h2 className={classNames(BoxTitle, className)} {...otherProps}>
      {children}
    </h2>
  );

  static Body = ({ children, className, ...otherProps }) => (
    <div className={classNames(BoxBody, className)} {...otherProps}>
      {children}
    </div>
  );

  static Option = ({ children, className, ...otherProps }) => (
    <div className={classNames(BoxOption, className)} {...otherProps}>
      {children}
    </div>
  );

  render() {
    const { className, center, padding, children, ...otherProps } = this.props;

    let classNameToUse = classNames(BoxBase, {
      [this.props.className]: !!className,
      [BoxCenter]: center,
      [BoxPadding]: padding,
    });

    return (
      <div className={classNameToUse} {...otherProps}>
        {children}
      </div>
    );
  }
}
