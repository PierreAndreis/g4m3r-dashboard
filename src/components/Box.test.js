import React from "react";

import Box from "./Box";
import { render } from "react-testing-library";

test("it mounts", () => {
  const { container } = render(<Box>
    <Box.Title>Title!</Box.Title>
    <Box.Body>Body!</Box.Body>
    <Box.Option>
      <div>Label</div>
      <div>Select</div>
    </Box.Option>
  </Box>);

  expect(container.firstChild).toMatchSnapshot();
});

test("Option throws if children is not of div type", () => {
  jest.spyOn(console, 'error');
  // Surpress console.error
  global.console.error.mockImplementation(() => {});

  expect(() => render(<Box.Option><select /></Box.Option>)).toThrow()
  global.console.error.mockRestore();
})
