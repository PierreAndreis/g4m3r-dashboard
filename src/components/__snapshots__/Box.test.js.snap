// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box it mounts 1`] = `
.emotion-3 {
  background: white;
  width: 350px;
  height: auto;
  box-shadow: 0 10px 40px 0 rgba(0,0,0,0.1);
  border-radius: 3px;
  box-sizing: border-box;
  margin: 5px;
  position: relative;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 18px;
  color: #4a4a4a;
  margin: 0 10px 10px;
}

.emotion-1 {
  width: 100%;
  box-sizing: border-box;
}

.css-nejrqv .emotion-1 > div {
  padding: 5px 15px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  padding: 20px;
}

.emotion-2 > div:nth-child(1) {
  -webkit-flex: 1.3;
  -ms-flex: 1.3;
  flex: 1.3;
  margin: 10px 0;
}

.emotion-2:nth-child(even) {
  background: rgba(0,0,0,0.03);
}

<div
  class="emotion-3"
>
  <h2
    class="emotion-0"
  >
    Title!
  </h2>
  <div
    class="emotion-1"
  >
    Body!
  </div>
  <div
    class="emotion-2"
  >
    <div>
      Label
    </div>
    <div>
      Select
    </div>
  </div>
</div>
`;
